lib/rex/post/meterpreter/command_mapper.rb

```diff
@@ -108,25 +109,31 @@ def self.get_commands(*extensions)
   # @param Integer value The value of the TLV type to retrieve the TLV type names for.
   # @return [Array<Symbol>] An array of symbols of all TLV types that are defined with the value. Can be empty.
   def self.get_tlv_names(value)
-    return @@cached_tlv_types[value] unless @@cached_tlv_types[value].nil? || @@cached_tlv_types[value].empty?
+    @@tlv_cache_mutex.synchronize_read do
```

Comment:
What impacts could this have on throughput?

Reply:
Performance shouldn't be an issue here, as exclusive write access will only be needed once to hydrate the cache at the start, from that point onwards it's purely reader access - which is the optimised codepath

That aside; I've created a separate issue to fix the thread safety issue in the reader/writer lock itself: https://github.com/rapid7/rex-core/issues/27